server/app/repository/ProgramRepository.java

```diff
@@ -66,6 +68,17 @@ public Program updateProgramSync(Program program) {
     return program;
   }
 
+  public ImmutableSet<String> getAllProgramNames() {
+    ImmutableSet.Builder<String> names = ImmutableSet.builder();
+    List<SqlRow> rows = database.sqlQuery("SELECT DISTINCT name FROM programs").findList();
```

Comment:
We don't use raw sql a lot that I've seen, can we stick with the DAO unless it can't meet our need here?

Reply:
I'm not sure how to do this otherwise, everything I try wants to return a list of `Program`s

Reply:
ah ok, I'd thought it should be easy enough but we can figure it out later.

Reply:
this returns a `Set<Program>`